### PR TITLE
feat: add Boba finalizer

### PR DESCRIPTION
**Pull request by nicholaspai**
Boba's fork of Optimism's SDK isn't published on NPM so I forked a repo under @across-protoocl/boba:packages/sdk

### CODE REVIEW

File: src/finalizer/index.ts [hunk 1]
```diff
@@ -167,22 +234,12 @@ async function updateFinalizerClients(clients: Clients) {
 }
 
 export class FinalizerConfig extends DataworkerConfig {
-  readonly optimisticRollupFinalizationWindow: number;
```
Comment (nicholaspai) [inline src/finalizer/index.ts:170]: We don't override these window settings in prod we should just default them as we don't see these windows changing anytime soon

File: src/finalizer/utils/optimism.ts [hunk 2]
```diff
@@ -1,27 +1,38 @@
...
-    l1SignerOrProvider: hubSigner.connect(new ethers.providers.JsonRpcProvider(getNodeUrlList(1)[0])),
-    l2SignerOrProvider: hubSigner.connect(new ethers.providers.JsonRpcProvider(getNodeUrlList(CHAIN_ID)[0])),
-  });
+export function getOptimismClient(chainId: OVM_CHAIN_ID, hubSigner: Wallet): OVM_CROSS_CHAIN_MESSENGER {
+  if (chainId === 288) {
+    return new bobaSDK.CrossChainMessenger({
+      l1ChainId: 1,
+      l1SignerOrProvider: hubSigner.connect(new ethers.providers.JsonRpcProvider(getNodeUrlList(1)[0])),
+      l2SignerOrProvider: hubSigner.connect(new ethers.providers.JsonRpcProvider(getNodeUrlList(chainId)[0])),
+      contracts: { ...bobaSDK.CONTRACT_ADDRESSES[1] }, // Override with Boba system addresses
```
Comment (nicholaspai) [inline src/finalizer/utils/optimism.ts:17]: This is the main change to the optimism SDK, we override the `contracts` map of addresses and contract names, as Boba added new contracts

Comment (mrice32) [reply src/finalizer/utils/optimism.ts:17]: Were you able to avoid the try-catch? This seems to work pretty well.

**Review by mrice32 — APPROVED** [review]
This looks great! Awesome work -- very fast timeline.

**Comment by nicholaspai** [comment]
> This looks great! Awesome work -- very fast timeline.

Unfortunately no, it the BobaSDK.CrossChainMessenger tries to load a hardcoded list of contract addresses like L1CrossDomainMessengerFast which doesn't exist on OptimismSDK

```sh
Error: Unable to find artifact for contract: L1CrossDomainMessengerFast
    at Object.getContractDefinition (/Users/nicholaspai/UMA/relayer-v2/node_modules/@across-protocol/boba-sdk/node_modules/@eth-optimism/contracts/dist/contract-defs.js:9:15)
    at getContractInterface (/Users/nicholaspai/UMA/relayer-v2/node_modules/@across-protocol/boba-sdk/node_modules/@eth-optimism/contracts/dist/contract-defs.js:15:32)
    at getOEContract (/Users/nicholaspai/UMA/relayer-v2/node_modules/@across-protocol/boba-sdk/src/utils/contracts.ts:278:25)
    at getAllOEContracts (/Users/nicholaspai/UMA/relayer-v2/node_modules/@across-protocol/boba-sdk/src/utils/contracts.ts:323:46)
    at new CrossChainMessenger (/Users/nicholaspai/UMA/relayer-v2/node_modules/@across-protocol/boba-sdk/src/cross-chain-messenger.ts:103:39)
    at getOptimismClient (/Users/nicholaspai/UMA/relayer-v2/src/finalizer/utils/optimism.ts:36:12)
    at finalize (/Users/nicholaspai/UMA/relayer-v2/src/finalizer/index.ts:176:52)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.runFinalizer [as finalizer] (/Users/nicholaspai/UMA/relayer-v2/src/finalizer/index.ts:264:9)
    at async run (/Users/nicholaspai/UMA/relayer-v2/index.ts:40:9)
```

**Pull request merged**